session/storage_memory.go

```diff
@@ -81,3 +81,13 @@ func (storage *StorageMemory) Remove(id ID) {
 	defer storage.lock.Unlock()
 	delete(storage.sessions, id)
 }
+
+// RemoveForProposal removes all sessions with given proposal id
+func (storage *StorageMemory) RemoveForProposal(proposalID int) {
```

Comment:
Why filtering by proposalId, it should search by unique index (like `session.id`) and there is a function for that already `storage.Remove(id)`

Reply:
When stopping service, we need to find sessions which belong to this service.
I've done this outside of `StorageMemory` at first, but then that method was feature-envy. So I've moved that method here.